doc/install.rst

```diff
@@ -1,19 +1,73 @@
+.. _installation:
 Installation
 ************
 
+.. note:: :mod:`message_ix_models` requires :mod:`message_ix` to run. Please ensure your system has `their required dependencies <https://docs.messageix.org/en/stable/install.html#install-system-dependencies>`_ installed.
+
+:mod:`message_ix_models` is structured as a Python package and is published to the PyPI public code repository. Hence, there are two options for the installation:
+
+
 From PyPI
-=========
+---------
+
+This option is only recommended for users who do not wish to make any changes to the source code.
 
 1. Run::
 
-    $ pip install message-ix-models
+    $ pip install message-ix-models[<extra_dependencies>]
 
 
 From source
-===========
+-----------
+
+Use this option if you intend to make changes to the source code. 
+We value your contributions via pull requests to `the main repository <https://github.com/iiasa/message-ix-models>`_. Please consider `contributing to development <https://docs.messageix.org/en/latest/contributing.html>`_ your changes.
```

Comment:
This is a direct URL link:
```rest
`contributing to development <https://docs.messageix.org/en/latest/contributing.html>`_
```

Wherever possible we use [intersphinx links](https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html), which are more reliable:
```rest
:doc:`message-ix:contributing`
```

- The prefixes before the `:` can be viewed in conf.py: https://github.com/iiasa/message-ix-models/blob/583d3e90a8ad5a8afcd28f4d7a1683e883c4d9ee/doc/conf.py#L134-L150
- The [docs show](https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html#showing-all-links-of-an-intersphinx-mapping-file) how to retrieve and inspect the list of :doc:, :ref: etc. targets available from a given other project.

Reply:
The command to retrieve the targets doesn't readily work for me. If I pass a directory to it, it gives me an `IsADirectoryError`; if I pass in a single file, I receive varying `ValueError`s because the inventory header is invalid; and the same is true if I pass in a URL to a page of our docs.
I guess I'll look up targets manually.

Reply:
Notice that in the example in the Sphinx docs, `…/objects.inv` is appended. This happens automatically during `sphinx-build`, but not for the helper command.

Reply:
Right, it needs to be the `objects.inv` file that I can find in e.g. `ixmp/doc/_build/`.

Reply:
If I understand correctly, intersphinx tries to resolve an undefined reference like `:doc:'installation'` by going through all other linked sources, so I tried following the [official guide](https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html#explicitly-reference-external-objects) and linked external reference in this way: `:external+message-ix:doc:'installation'`.